lib/shared/sse-connection/src/lib/SSEConnection.ts

```diff
@@ -0,0 +1,49 @@
+import type { DVCLogger } from '@devcycle/types'
+
+export class SSEConnection {
+    private connection?: EventSource
+
+    constructor(
+        private url: string,
+        private onMessage: (message: unknown) => void,
+        private logger: DVCLogger,
+    ) {
+        this.openConnection()
+    }
+
+    private openConnection() {
+        if (typeof EventSource === 'undefined') {
+            this.logger.warn(
+                'StreamingConnection not opened. EventSource is not available.',
+            )
+            return
+        }
+        this.connection = new EventSource(this.url, { withCredentials: true })
+        this.connection.onmessage = (event) => {
+            this.onMessage(event.data)
+        }
+        this.connection.onerror = () => {
+            this.logger.warn(
+                'StreamingConnection warning. Connection failed to establish.',
+            )
+        }
+        this.connection.onopen = () => {
+            this.logger.debug('StreamingConnection opened')
+        }
+    }
+
+    isConnected(): boolean {
+        return this.connection?.readyState === this.connection?.OPEN
+    }
+
+    reopen(): void {
+        if (!this.isConnected()) {
+            this.close()
+            this.openConnection()
```

Comment:
Is a connection that's not connected required to be closed?
Do we care about connected sdks being told to close and then reconnect?

Reply:
yea, not sure yet, will figure that out when I try and integrate into NodeJS.